src/OrbitGl/GpuDebugMarkerTrack.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "GpuDebugMarkerTrack.h"
+
+#include <absl/time/time.h>
+
+#include <memory>
+
+#include "App.h"
+#include "Batcher.h"
+#include "CoreUtils.h"
+#include "GlCanvas.h"
+#include "GlUtils.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ThreadConstants.h"
+#include "OrbitClientModel/CaptureData.h"
+#include "TimeGraph.h"
+#include "TimeGraphLayout.h"
+#include "TimerChain.h"
+#include "TriangleToggle.h"
+#include "absl/strings/str_format.h"
+
+using orbit_client_protos::TimerInfo;
+
+GpuDebugMarkerTrack::GpuDebugMarkerTrack(CaptureViewElement* parent, TimeGraph* time_graph,
+                                         TimeGraphLayout* layout, OrbitApp* app,
+                                         const CaptureData* capture_data, uint32_t intent)
+    : TimerTrack(parent, time_graph, layout, app, capture_data, intent) {
+  SetLabel("Debug Markers");
+  transparent_ = true;
+  string_manager_ = app->GetStringManager();
+
+  // Gpu tracks are collapsed by default.
+  collapse_toggle_->SetState(TriangleToggle::State::kCollapsed,
+                             TriangleToggle::InitialStateUpdate::kReplaceInitialState);
+}
+
+Color GpuDebugMarkerTrack::GetTimerColor(const TimerInfo& timer_info, bool is_selected,
+                                         bool is_highlighted) const {
+  CHECK(timer_info.type() == TimerInfo::kGpuDebugMarker);
+  const Color kInactiveColor(100, 100, 100, 255);
+  const Color kSelectionColor(0, 128, 255, 255);
+  if (is_highlighted) {
+    return TimerTrack::kHighlightColor;
+  }
+  if (is_selected) {
+    return kSelectionColor;
+  }
+  if (!IsTimerActive(timer_info)) {
+    return kInactiveColor;
+  }
+  if (timer_info.has_color()) {
+    CHECK(timer_info.color().red() < 256);
+    CHECK(timer_info.color().green() < 256);
+    CHECK(timer_info.color().blue() < 256);
+    CHECK(timer_info.color().alpha() < 256);
```

Comment:
Since TimerInfo is simply a proto, we can only check this on the usage sides (unless we create a wrapper).

Re: ColorFromProtoColor, I kind of agree. However, my question would be where to put it. Currently we do have a couple of static GetColor functions in TimeGraph. But I would be afraid of adding more noise to the timegraph. Refactoring those out would be out of scope (as the scope of this PR is already a quite large).